pennylane/devices/default_qubit.py

```diff
@@ -604,25 +604,8 @@ def density_matrix(self, wires):
             array[complex]: complex tensor of shape ``(2 ** len(wires), 2 ** len(wires))``
             representing the reduced density matrix.
         """
-        dim = self.num_wires
-        state = self._pre_rotated_state
-
-        # Return the full density matrix by using numpy tensor product
-        if wires == self.wires:
-            density_matrix = self._tensordot(state, self._conj(state), axes=0)
-            density_matrix = self._reshape(density_matrix, (2 ** len(wires), 2 ** len(wires)))
-            return density_matrix
-
-        complete_system = list(range(0, dim))
-        traced_system = [x for x in complete_system if x not in wires.labels]
-
-        # Return the reduced density matrix by using numpy tensor product
-        density_matrix = self._tensordot(
-            state, self._conj(state), axes=(traced_system, traced_system)
-        )
-        density_matrix = self._reshape(density_matrix, (2 ** len(wires), 2 ** len(wires)))
-
-        return density_matrix
+        wires = wires.tolist()
+        return qml.math.state_to_density_matrix(self.state, wires=wires)
```

Comment:
Devices may have dedicated `self.state` definitions, maybe it's best to keep using `self._pre_rotated_state` directly here.
```suggestion
        return qml.math.state_to_density_matrix(self._pre_rotated_state, wires=wires)
```